packages/macro/src/index.ts

```diff
@@ -51,15 +58,17 @@ function macro({ references, state, babel }: MacroParams) {
     }
   })
 
+  const stripNonEssentialProps = process.env.NODE_ENV == "production" && !opts.extract;
+
   jsNodes.filter(isRootPath(jsNodes)).forEach((path) => {
     if (alreadyVisited(path)) return
-    const macro = new MacroJS(babel, { i18nImportName })
+    const macro = new MacroJS(babel, { i18nImportName, stripNonEssentialProps })
```

Comment:
I also encountered that only JSX macro stripping non essential props.  JS macro doesn't have such logic, we need to fix it next iterations